state/validators.md

```diff
@@ -0,0 +1,38 @@
+# Validator Selection Scheme
+
+## Abstract
+
+This document elucidates the validator selection mechanism as outlined in the [Rollkit repository](https://github.com/rollkit/rollkit). The scheme establishes an initial set of validators at the genesis of the blockchain, then leverages a priority-based system to determine the proposer of new blocks, ensuring a decentralized and fair block proposal mechanism.
+
+## Protocol/Component Description
+
+The protocol initializes with a predefined set of validators as stated in the `GenesisDoc`. These validators are part of the `GenesisValidator` struct which includes their address, public key, voting power, and name. Post genesis, the validator set might change, and is managed through Tendermint's `ValidatorSet` and `State` structures.
+
+The `ValidatorSet` represents a set of `Validator` at a given height, ordered by their voting power in descending order and secondary by their address in ascending order. The `ProposerPriority` of each validator and the designated proposer changes every round, upon calling `.IncrementProposerPriority()`.
+
+In the `State` struct, three kinds of validator sets are defined: `Validators`, `NextValidators`, and `LastValidators`. The `NextValidators` set is generated by copying the current `Validators` set and incrementing the proposer priority, as seen in the `updateState` function.
+
+The next proposer is ascertained based on the priority and verified in `manager.go` through the `IsProposer` method. Only the identified proposer is permitted to publish new blocks, ensuring a single block proposer at any given time.
+
+Furthermore, the block header contains hashes of the current and next validator sets (termed as aggregators) which are assigned during block creation and verified during block verification, ensuring continuity and agreement on the validator sets across the network.
+
+## Message Structure/Communication Format
+
+The primary structures encompassing validator information include `GenesisValidator`, `ValidatorSet`, and `State`. These structures are serialized using JSON and the communication occurs through the Tendermint's ABCI (Application BlockChain Interface) as seen in `updateState` method where `ValidatorUpdates` from the response is used to update the state's validator sets.
```

Comment:
Instead of `Tendermint` here, `CometBFT` should be used since Rollkit imports `CometBFT` currently, not `Tendermint`.

Reply:
Updated.